javascript-engine/src/index.ts

```diff
@@ -0,0 +1,96 @@
+import init, { Engine } from '@unleash/yggdrasil-wasm'
+import { checkResponse, readResponse } from './wasm-interop'
+import { Strategies } from './strategies'
+import type {
+  Context,
+  FeatureDefinition,
+  MetricsBucket,
+  State,
+  Variant,
+  Strategy,
+  DeltaState
+} from './types'
+export * from './types'
```

Comment:
I don't think these types are sane enough for external usage. They are mostly internal and incomplete.

 Should we leave them out for now until we figure out the exact types we want to expose?

```suggestion
```

Reply:
You'll definitely need some of them but I think that's sane. I support having a public API that's as narrow as possible!